packages/wallets/solflare/src/adapter.ts

```diff
@@ -166,6 +170,29 @@ export class SolflareWalletAdapter extends BaseMessageSignerWalletAdapter {
         }
     }
 
+    private _detectWallet = async () => {
+        const Solflare = (await import('@solflare-wallet/sdk')).default;
```

Comment:
Hmm, I can't accept this change. All adapters that import third party libraries do so when connecting the wallet, which requires a user action or an app to explicitly call `connect`.

This change is causing the library to be loaded immediately when the adapter's constructor is called. It shoudn't be necessary to load a library to detect the extension, so please refactor.

The code used in the other adapters should work here prettymuch out of the box.